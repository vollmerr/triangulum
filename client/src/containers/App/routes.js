import CrawlForm from '../CrawlForm';
import CrawlGraph from '../CrawlGraph';
import CrawlHistory from '../CrawlHistory';

const routes = {
  form: {
    key: 'form',
    exact: true,
    name: 'New',
    path: '/',
    icon: 'plus',
    component: CrawlForm,
  },
  graph: {
    key: 'graph',
    name: 'Graph',
    path: '/graph',
    icon: 'chart area',
    component: CrawlGraph,
  },
  history: {
    key: 'history',
    name: 'History',
    path: '/history',
    icon: 'history',
    component: CrawlHistory,
  },
};

export default Object.values(routes);
