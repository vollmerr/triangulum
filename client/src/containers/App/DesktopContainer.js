import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import {
  Container,
  Icon,
  Menu,
  Responsive,
  Segment,
} from 'semantic-ui-react';

import routes from './routes';

export const Logo = () => (
  <Menu.Item as={Link} to={'/'}>
    <Icon name={'sitemap'} style={{ transform: 'rotate(0.5turn)' }} />
    <span>riangulum</span>
  </Menu.Item>
);

export const Links = ({ path }) => (
  routes.map(route => (
    <Menu.Item
      key={route.key}
      icon={route.icon}
      content={route.name}
      as={Link}
      to={route.path}
      active={path === route.path}
    />
  ))
);

export class DesktopContainer extends Component {
  state = {}

  render() {
    const { children, location } = this.props;

    return (
      <Responsive minWidth={Responsive.onlyTablet.minWidth} id={'desktop-view'}>

        <Segment
          inverted
          textAlign='center'
          style={{ minHeight: 40, padding: '1em 0em', margin: '0 0 2em 0' }}
          color={'violet'}
          vertical
        >
          <Container>
            <Menu
              fixed={'top'}
              inverted
              pointing
              secondary
              color={'violet'}
            >
              <Logo />
              <Menu.Item position={'right'} />
              <Links path={location.pathname} />
            </Menu>
          </Container>
        </Segment>

        {children}
      </Responsive>
    );
  }
}

export default withRouter(DesktopContainer);
