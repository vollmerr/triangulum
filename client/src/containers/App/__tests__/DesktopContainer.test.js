import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from 'semantic-ui-react';

import routes from '../routes';
import { DesktopContainer, Logo, Links } from '../DesktopContainer';

let wrapper;
const props = {
  location: {
    pathname: '/test-path',
  },
};
const Child = () => <div>children...</div>;

describe('Logo', () => {
  it('should render correctly', () => {
    wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Links', () => {
  it('should map all the routes as menu items', () => {
    wrapper = shallow(<Links />);
    const items = wrapper.find(Menu.Item);
    expect(items.length).toEqual(routes.length);
  });
});

describe('DesktopContainer', () => {
  beforeEach(() => {
    wrapper = shallow((
      <DesktopContainer {...props}>
        <Child />
      </DesktopContainer>
    ));
  });

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the chidlren', () => {
      expect(wrapper.find(Child).length).toEqual(1);
    });
  });
});
