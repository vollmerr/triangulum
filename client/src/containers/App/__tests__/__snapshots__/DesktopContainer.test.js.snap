// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopContainer render should render correctly 1`] = `
<Responsive
  getWidth={[Function]}
  id="desktop-view"
  minWidth={768}
>
  <Segment
    color="violet"
    inverted={true}
    style={
      Object {
        "margin": "0 0 2em 0",
        "minHeight": 40,
        "padding": "1em 0em",
      }
    }
    textAlign="center"
    vertical={true}
  >
    <Container>
      <Menu
        color="violet"
        fixed="top"
        inverted={true}
        pointing={true}
        secondary={true}
      >
        <Logo />
        <MenuItem
          position="right"
        />
        <Links
          path="/test-path"
        />
      </Menu>
    </Container>
  </Segment>
  <Child />
</Responsive>
`;

exports[`Logo should render correctly 1`] = `
<MenuItem
  as={[Function]}
  to="/"
>
  <Icon
    as="i"
    name="sitemap"
    style={
      Object {
        "transform": "rotate(0.5turn)",
      }
    }
  />
  <span>
    riangulum
  </span>
</MenuItem>
`;
